test/poly/spoly-test.jl

```diff
@@ -414,39 +414,39 @@ end
    @test jf == x^3
 end
 
-@testset "spoly.test_spoly_factor..." begin
-   R1 , (x, y, z, w) = Singular.PolynomialRing(Singular.QQ,
-   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
-   f1 = 113*(2*y^7 + w^2)^3*(1 + x)^2*(x + y*z)^2
-
-   R2 , (x, y, z, w) = Singular.PolynomialRing(Singular.ZZ,
-   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
-   f2 = 123*(57*y^3 + w^5)^3*(x^2 + x+1)^2*(x + y*z)^2
-
-   R3 , (x, y, z, w) = Singular.PolynomialRing(Singular.Fp(3),
-   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
-   f3 = 7*(y^3 + w^3)*(1 + x)^2*(x + y*z)^2
-
-   for f in [f1, f2, f3]
-      #Test factor
-      F = factor(f)
-      g = F.unit
-      for p in keys(F.fac)
-         g = g*p^F[p]
-      end
-     @test f == g
-
-     #Test factor_squarefree over QQ and Fp
-      if typeof(f.parent) != PolyRing{n_Z}
-         F = factor_squarefree(f)
-         g = F.unit
-         for p in keys(F.fac)
-            g = g*p^F[p]
-         end
-         @test f == g
-      end
-   end
-end
+#@testset "spoly.test_spoly_factor..." begin
+#   R1 , (x, y, z, w) = Singular.PolynomialRing(Singular.QQ,
+#   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
+#   f1 = 113*(2*y^7 + w^2)^3*(1 + x)^2*(x + y*z)^2
+#
+#   R2 , (x, y, z, w) = Singular.PolynomialRing(Singular.ZZ,
+#   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
+#   f2 = 123*(57*y^3 + w^5)^3*(x^2 + x+1)^2*(x + y*z)^2
+#
+#   R3 , (x, y, z, w) = Singular.PolynomialRing(Singular.Fp(3),
+#   ["x", "y", "z", "w"]; ordering=:negdegrevlex)
+#   f3 = 7*(y^3 + w^3)*(1 + x)^2*(x + y*z)^2
+#
+#   for f in [f1, f2, f3]
+#      #Test factor
+#      F = factor(f)
+#      g = F.unit
+#      for p in keys(F.fac)
+#         g = g*p^F[p]
+#      end
+#     @test f == g
+#
+#     #Test factor_squarefree over QQ and Fp
+#      if typeof(f.parent) != PolyRing{n_Z}
+#         F = factor_squarefree(f)
+#         g = F.unit
+#         for p in keys(F.fac)
+#            g = g*p^F[p]
+#         end
+#         @test f == g
+#      end
+#   end
+#end
```

Comment:
I assume this code is commented out because of what  you mentioned in the PR description? I.e.:
> They are failing for ZZ with a wrong result.
> They are failing for ZZ/3 because NTL is missing

(I actually don't see ZZ/3 mentioned in the test code above, perhaps this is an indirect problem, and/or simply not covered by a test)?

Anyway, please add a comment before the commented out test code explaining why it is commented out, so that we know what's going on for revisiting it later? And perhaps also submit an issue about the... issue  here, so that we remember to look into it at some point (whatever that means: updating/improving Singular, or something else, doesn't matter).

Otherwise: great job, thank you!

Reply:
In the code it is called `Fp(3)`.

I don't have a local Singular laying around. Maybe @hannes14 can have a look.

Speaking of it, should we have used `--with-ntl=no` when building the binaries?

Reply:
Ah, I missed the `Fp(3)`, thank you. So, just to check that I understood it correctly: the above test works for f1, gives a wrong result for f2, and fails for Z/3Z because Singular complains about NTL missing? Sounds indeed like something @hannes14 (or perhaps also @tthsqe12) ought to look into at some point :-)

Reply:
And I don't think we need that --with-ntl=no -- it detects that there is no NTL, so the "no" is implied from there.

Reply:
So for f2, it seems the content of the polynomial, 123, is omitted: 
```
julia> R2 , (x, y, z, w) = Singular.PolynomialRing(Singular.ZZ,
       ["x", "y", "z", "w"]; ordering=:negdegrevlex)
(Singular Polynomial Ring (ZZ),(x,y,z,w),(ds(4),C), spoly{n_Z}[x, y, z, w])

julia> f2 = 123*(57*y^3 + w^5)^3*(x^2 + x+1)^2*(x + y*z)^2;

julia> factor(f2)
1 * (57*y^3+w^5)^3 * (x+y*z)^2 * (1+x+x^2)^2
```
So this is clearly a Singular bug, hopefully easy to resolve for @hannes14 

For f3, we get:
```
julia> F = factor(f)
multivariate factorization depends on NTL(missing)
signal (6): Abort trap: 6
in expression starting at REPL[31]:1
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 27262687 (Pool: 27254381; Big: 8306); GC: 30
Abort trap: 6
```
Which is not good: besides the missing factorization, I am worried that this leads to a segfault, not just an exception, so that seems like two bugs to me.

Reply:
Because of segfault, I thought that we might be configuring Singular wrongly, but apparently this is the intented behavior.

Reply:
signal 6 is not segfault but a call to abort, changed in https://github.com/Singular/Singular/commit/a71eabdfff523e0626ba9232736fadfa4ae937eb.
I cannot reproduce the f2-example in Singular.

Reply:
Sorry, you are right.

Regarding the f2-example. We are currently using https://github.com/Singular/Singular/commit/7ee144671ad6b1f2bb437bbbeb8ee7559f5029cb for the binaries. Could this have been fixed since then?

Is it a good idea to build new binaries or are there more important changes coming in the next days?

Reply:
Holiday till September, so do not expect many changes in the next days in Singular.
Using https://github.com/Singular/Singular/commit/7ee144671ad6b1f2bb437bbbeb8ee7559f5029cb misses only the
adaption to flint 2.6.3, so the bug for the f2-example is probably somethere else.

Reply:
@hannes14 I can reproduce the f2 problem in Singular `spielwiese` (when built against FLINT 2.6.3, with NTL disabled). I submitted this as https://github.com/Singular/Singular/issues/1025

Reply:
Anyway, happy holidays ;-) and perhaps @tthsqe12 can look into it (at least @wdecker told me that that would be the canonical thing)

Reply:
I cancelled the new Singular_jll build, so from my part this is good to go.